contrib/node/src/python/pants/contrib/node/tasks/node_resolve.py

```diff
@@ -160,4 +165,9 @@ def execute(self):
                                                   resolve_locally=True,
                                                   install_optional=True,
                                                   frozen_lockfile=False)
-          node_paths_local.resolved(target, results_dir)
+          # If the target is synthetic, the node_paths_local is to be resolved with the derived
```

Comment:
This feels like it's undoing what was done above where I commented on the `isinstanceof(NodePathsLocal)`... it's not clear why these checks are relevant if we are properly filtering to the generated target type.

Some of this feels like maybe the thrift target type is extending `NodeModule`, but shouldn't be? And so it needs to be filtered out or handled by a bunch of Tasks that are encountering it later?

Reply:
node_thrift_module should be in every way a node_module, with the exception of the code generation. For now, we remove certain features by blocking user access to creating their own `package.json` and `yarn.lock` file until we can support those features such as build scripts and bundling.

Reply:
> This feels like it's undoing what was done above where I commented on the isinstanceof(NodePathsLocal)... it's not clear why these checks are relevant if we are properly filtering to the generated target type.

The main need for this check is that during a `node-install` task, the target dependencies are found using relative paths. In this specific case, the resolved path will be a `node_module` type because the installed location will exist in the pants.d. The issue is that dependencies are discovered via relative path from `source` and cause a miss in node_paths cache.